services/task-svc/internal/ward/ward.go

```diff
@@ -237,4 +231,73 @@ func (s ServiceServer) GetWardOverviews(ctx context.Context, _ *pb.GetWardOvervi
 	})
 
 	return &pb.GetWardOverviewsResponse{Wards: resWards}, err
+
+}
+
+func (ServiceServer) GetWardDetails(ctx context.Context, req *pb.GetWardDetailsRequest) (*pb.GetWardDetailsResponse, error) {
+
+	wardID, err := uuid.Parse(req.Id)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	wardRepository := wardmodels.NewWardRepositoryWithDB(hwgorm.GetDB(ctx))
+	ward, err := wardRepository.GetWardById(wardID)
+	if err != nil {
+		return nil, err
```

Comment:
We should handle the database errors correctly like below. It looks like this was also forgotten in other previously implemented methods (by me :)).
```json
if hwgorm.IsOurFault(err) {
	return nil, status.Error(codes.Internal, err.Error())
} else {
	return nil, status.Error(codes.InvalidArgument, "id not found")
}
```